lib/icinga/clusterevents.cpp

```diff
@@ -1252,7 +1253,8 @@ Value ClusterEvents::UpdateExecutionsAPIHandler(const MessageOrigin::Ptr& origin
 
 	ObjectLock oLock (checkable);
 
-	if (origin->FromZone && !origin->FromZone->CanAccessObject(checkable)) {
+	if (origin->FromZone && !origin->FromZone->CanAccessObject(checkable) &&
+		endpoint->GetName() != host->GetName()) {
```

Comment:
`endpoint` is obtained by `origin->FromClient->GetEndpoint()`, i.e. the client from which the current node received the cluster message. In case the message is relayed by a satellite, this is not the node from which the message originated but the one that relayed it.